### ISSUE TITLE
Updating Text/Mechanic

### ISSUE THREAD

**Issue by reidospalas** [stoneharry/Spell-Editor-GUI-V2#50]
When i open the Editor and click in some spell, it stuck in a infinity loading screen, sometimes "Updating text control's" and sometimes "Updating category & dispel & mechanic"

thank you :)

**Comment by stoneharry** [comment]
Which spell id does this happen with?

**Comment by reidospalas** [comment]
All spells is getting this error, and the ICON bug still up, Client Crash when try using custom ICONS, i don't know if theres a limit but i'm tryin' use 17k Icons and is crashing while load like 80-90% of them

**Comment by stoneharry** [comment]
It sounds like bad configuration.

Do you have an "error.txt" present in the same directory as the spell editor program? What is the contest of this file?

**Comment by stoneharry** [comment]
I found two unsupported BLP images when testing with another users data set that causes the program to crash when going to the 'Icon' tab.

```Exception thrown: 'System.IndexOutOfRangeException' in SereniaBLPLib
Error loading image, unsupported BLP format: INTERFACE\ICONS\thumbup.blp
Index was outside the bounds of the array.
System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at SereniaBLPLib.BlpFile.getPictureUncompressedByteArray(Int32 MipmapLevel)
   at SereniaBLPLib.BlpFile.getImageBytes(Int32 MipmapLevel)
   at SereniaBLPLib.BlpFile.getBitmap(Int32 MipmapLevel)
   at SpellEditor.Sources.DBC.SpellIconDBC.<UpdateMainWindowIcons>d__10.MoveNext() in D:\WoW Resources\Spell-Editor-GUI-V2\SpellGUIV2\Sources\DBC\SpellIconDBC.cs:line 212
Error loading image, unsupported BLP format: INTERFACE\ICONS\thumbsdown.blp
Index was outside the bounds of the array.
System.IndexOutOfRangeException: Index was outside the bounds of the array.
   at SereniaBLPLib.BlpFile.getPictureUncompressedByteArray(Int32 MipmapLevel)
   at SereniaBLPLib.BlpFile.getImageBytes(Int32 MipmapLevel)
   at SereniaBLPLib.BlpFile.getBitmap(Int32 MipmapLevel)```

I have fixed this in fcba523ddf576b376c1c6862595b7ae009013a99.

**Comment by stoneharry** [comment]
Release 1.6.2 should resolve these issues. Please test and reopen the ticket if this is still an issue.

https://github.com/stoneharry/Spell-Editor-GUI-V2/releases/tag/v1.6.2
